testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/jpa2lc/ClusteredJPA2LCCustomRegionTestCase.java

```diff
@@ -0,0 +1,397 @@
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.descriptor.api.Descriptors;
+import org.jboss.shrinkwrap.descriptor.api.spec.se.manifest.ManifestDescriptor;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.api.Authentication;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.UnknownHostException;
+
+import static org.jboss.as.controller.client.helpers.ClientConstants.ADD;
+import static org.jboss.as.controller.client.helpers.ClientConstants.ADDRESS;
+import static org.jboss.as.controller.client.helpers.ClientConstants.OP;
+import static org.jboss.as.controller.client.helpers.ClientConstants.OUTCOME;
+import static org.jboss.as.controller.client.helpers.ClientConstants.REMOVE_OPERATION;
+import static org.jboss.as.controller.client.helpers.ClientConstants.SUCCESS;
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.DEPLOYMENT_1;
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.DEPLOYMENT_2;
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.NODE_1;
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.NODE_2;
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.TWO_DEPLOYMENTS;
+import static org.jboss.as.test.clustering.cluster.AbstractClusteringTestCase.TWO_NODES;
+
+/**
+ * Test of clustered JPA 2nd level cache implemented by Infinispan using entity custom region.
+ * <p>
+ * In persistence.xml we add the following:
+ * <code>
+ * <property name="hibernate.cache.infinispan.entity.cfg" value="entity-replicated"/>
+ * </code>
+ * This has the effect of using the Infinispan cache "entity-replicated" as the template for entities's caches;
+ * </p>
+ * <p>
+ * In the standalone-ha.xml, we have an invalidation-cache named "entity" in the cache container "hibernate" of
+ * the Infinispan subsystem;
+ * By default, this cache would be used as the template for entities's caches;
+ * Here we are overriding this default creating a replicated-cache in the cache container "hibernate", and using it
+ * as template;
+ * </p>
+ * <p>
+ * We are using the entity {@link DummyEntityCustomRegion}, which also defines a custom region name to be used for
+ * its second level cache name:
+ * <code>
+ *
+ * @author Jan Martiska and Tommaso Borgato
+ * @Cache(usage = CacheConcurrencyStrategy.TRANSACTIONAL, region = DummyEntityCustomRegion.DUMMY_ENTITY_REGION_NAME)
```

Comment:
Idea shows me warning here (Wrong tag Cache)

Reply:
@javax.persistence.Cache don't have those attributes: @org.hibernate.annotations.Cache was used here; I copied it from the existing test

Reply:
I think that Idea warning means that it is wrong format javadoc